lib/MusicBrainz/Server/Controller/WS/js/Edit.pm

```diff
@@ -261,26 +266,34 @@ sub process_artist_credits {
             trim_string($name, 'name');
             trim_string($artist, 'name');
 
-            if (!$artist->{id} && is_guid($artist->{gid}))  {
-                push @artist_gids, $artist->{gid};
+            if (is_database_row_id($artist->{id}))  {
+                push @artist_ids, $artist->{id};
+            } elsif (is_guid($artist->{gid})) {
+                push @artist_ids, $artist->{gid};
```

Comment:
Why does this need to check for gid or id? The commit message doesn't talk about that and it's not obvious to me :)

Reply:
Updated the commit message, hope this helps:

> You'll notice that some incidental improvements have been made to `process_artist_credits` related to id/gid loading. Previously, we only checked for and loaded gids to fill in numeric ids where they were missing. Since we now want to fill in the name if empty for *all* artists, we load all artists, regardless of the type of id, and do that. This is in addition to filling in the numeric id as before -- and the new code supports an additional case where an artist submitted with only a numeric id will have a gid added too.